File: app/src/main/java/com/zack/enderplan/view/fragment/BaseFragment.java
package com.zack.enderplan.view.fragment;

import android.support.annotation.StringRes;
import android.support.v4.app.Fragment;
import android.widget.Toast;

public abstract class BaseFragment extends Fragment {

    public void showToast(@StringRes int msgResId) {
        Toast.makeText(getContext(), msgResId, Toast.LENGTH_SHORT).show();
    }

    protected void remove() {
        getFragmentManager().beginTransaction().remove(this).commit();
    }
}

File: app/src/main/kotlin/me/imzack/app/end/view/fragment/BaseFragment.kt
package me.imzack.app.end.view.fragment

import android.os.Bundle
import android.support.annotation.StringRes
import android.support.v4.app.Fragment
import android.widget.Toast

abstract class BaseFragment : Fragment() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        onInjectPresenter()
    }

    open fun onInjectPresenter() {

    }

    protected val fragmentName
        get() = javaClass.simpleName!!

    fun showToast(@StringRes msgResId: Int) {
        Toast.makeText(context, msgResId, Toast.LENGTH_SHORT).show()
    }

    protected fun remove() {
        fragmentManager!!.beginTransaction().remove(this).commit()
    }

    open fun exit() {
        remove()
    }
}
